### PR TITLE
Net app initial push

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by corbinhak** [comment]
This may be a big ask, but would you be able to redo the file structuring? The reason why this Github PR is so big, is that its seems everything from our front-end got also copied into your new backend folder(ClientApp is a duplicate), when it should only have the files relating to the backend.

File: SWEBackend/SWEBackend/API/RoomsAPI.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        public RoomsAPI(IRoomsController roomsController)
+            => _roomsController = roomsController;
+
+        [HttpGet]
+        public async Task<List<Room>> GetRoomsAsync(int? venueId)
+            => await _roomsController.GetRoomsAsync(venueId);
+
+        [HttpGet("{id}")]
+        public async Task<Room> GetRoomAsync(int id)
```
Comment (HeafieldLuke) [inline SWEBackend/SWEBackend/API/RoomsAPI.cs]: I don't think that we should express ID's as ints. I imagine there is some sort of `UUID` package that comes with this framework

File: SWEBackend/SWEBackend/ClientApp/src/App.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.App {
```
Comment (HeafieldLuke) [inline SWEBackend/SWEBackend/ClientApp/src/App.css]: Why is all the code from the React App copied and pasted here? It already exists.

File: SWEBackend/SWEBackend/API/RoomsAPI.cs (hunk 1 above)
Comment (ghost) [reply SWEBackend/SWEBackend/API/RoomsAPI.cs]: auto-incremented int is the default identity column for sql, no need for UUIDs

File: SWEBackend/SWEBackend/ClientApp/src/App.css (hunk 2 above)
Comment (ghost) [reply SWEBackend/SWEBackend/ClientApp/src/App.css]: This "backend" app is the entire hosted application, the "client" folder will be deleted and you guys will work out of the ClientApp folder within this sln

**Comment by JasonF2** [comment]
> This may be a big ask, but would you be able to redo the file structuring? The reason why this Github PR is so big, is that its seems everything from our front-end got also copied into your new backend folder(ClientApp is a duplicate), when it should only have the files relating to the backend.

File: SWEBackend/SWEBackend/API/RoomsAPI.cs (hunk 1 above)
Comment (HeafieldLuke) [reply SWEBackend/SWEBackend/API/RoomsAPI.cs]: It's the default, but it probably isn't an awesome idea.
https://devforth.io/blog/why-your-software-should-use-uuids

Comment (ghost) [reply SWEBackend/SWEBackend/API/RoomsAPI.cs]: I just mean its definitely overkill for this project, but I can change it.

File: SWEBackend/SWEBackend/Migrations/20210325204723_Initial.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                        .Annotation("Npgsql:ValueGenerationStrategy", NpgsqlValueGenerationStrategy.IdentityByDefaultColumn),
+                    ByteData = table.Column<byte[]>(type: "bytea", nullable: true)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_SWEFile", x => x.Id);
+                });
+
+            migrationBuilder.CreateTable(
+                name: "TimeSlots",
```
Comment (corbinhak) [inline SWEBackend/SWEBackend/Migrations/20210325204723_Initial.cs]: Sorry for not communicating this sooner, he wanted us to change the time slots to also include a duration field. Would we be able to add the logic here for it.

Comment (ghost) [reply SWEBackend/SWEBackend/Migrations/20210325204723_Initial.cs]: No worries, I must have missed that conversation. Cant that be calculated from the start and end time though?

Comment (corbinhak) [reply SWEBackend/SWEBackend/Migrations/20210325204723_Initial.cs]: Yes it can, wasn't sure where we wanted to perform that calculation.

Comment (ghost) [reply SWEBackend/SWEBackend/Migrations/20210325204723_Initial.cs]: Probably client side but we can add it to the api endpoint on the server if needed down the line easily enough

**Comment by ghost** [comment]
> This may be a question for the rest of the team to look at. Is SpeakerPhoto meant to be an actual ID number? If I’m not mistaken I think we were considering having an actual photo for each of the speakers and if so this field may have meant to be a file or link.

SpeakerPhotoId is a foreign key to the photo (SWEFile object)

File: SWEPortal/SWEPortal/Startup.cs [hunk 4]
```diff
@@ -25,9 +25,13 @@ public Startup(IConfiguration configuration)
...
         {
-            services.AddEntityFrameworkNpgsql().AddDbContext<ApplicationDbContext>(opt =>
-                    opt.UseNpgsql(Configuration.GetConnectionString("DefaultConnection")));
-
+            if (!string.IsNullOrWhiteSpace(Configuration.GetConnectionString("DefaultConnection")))
+                services.AddEntityFrameworkNpgsql().AddDbContext<ApplicationDbContext>(opt =>
+                        opt.UseNpgsql(Configuration.GetConnectionString("DefaultConnection")));
+            else
+                services.AddDbContext<ApplicationDbContext>(opt =>
+                        opt.UseSqlite(Configuration.GetSection("sqllitedb")?.Value ?? "Data Source=sweportal.db;"), ServiceLifetime.Transient);
```
Comment (HeafieldLuke) [inline SWEPortal/SWEPortal/Startup.cs:33]: Curious what the purpose of doing this is?

**Review by HeafieldLuke — APPROVED** [review]
Looks good to me, although I don't understand the framework well enough to catch any potential issues.

**Pull request merged**